pep-0517.txt

```diff
@@ -166,48 +166,71 @@ The build backend object is expected to have attributes which provide
 some or all of the following hooks. The common ``config_settings``
 argument is described after the individual hooks::
 
-  def get_build_requires(config_settings):
+  def build_sdist(source_directory, sdist_directory, config_settings):
+      ...
+
+MUST build an unpacked sdist from the ``source_directory`` and place it into the
+specified ``sdist_directory``. If the ``source_directory`` is already an
+unpacked sdist (as opposed to a VCS source tree) then it SHOULD simply copy the
+entire tree into ``sdist_directory``.
+
+This method only places the *contents* of the sdist file, without actually
+creating the final artifact. This is to allow flexibility in how the result is
+going to be used. A front end like pip, which is only going to immediately
+build a wheel from this directory can speed up the installation process by
+skipping a round trip through tar+gz. However a command like a hypothetical
+``twine sdist`` could call this method and then ultimately tar+gz up this
+directory to produce the sdist artifact.
+
+Mandatory.
+
+  def get_build_requires(source_directory, config_settings):
       ...
 
 This hook MUST return an additional list of strings containing PEP 508
 dependency specifications, above and beyond those specified in the
-``pyproject.toml`` file. Example::
+``pyproject.toml`` file. The ``source_directory`` parameter MUST be a path that
+points to the source tree that is being operated on. Example::
```

Comment:
> should this PEP also specify guidelines on handling CWD? like "if you os.chdir() somewhere else, make sure you go back to where you started"?

The idea is that the frontend would run these in a separate process, which would complete after running the hook, so there's no need for it to reset global state.